client/src/pages/ContractorSignUp.js

```diff
@@ -0,0 +1,148 @@
+import React, { useState } from "react";
+
+function ContractorSignUp() {
+	const [contractorDetails, setContractorDetails] = useState({
+		userType: "contractor",
+		firstName: "",
+		lastName: "",
+		email: "",
+		company: "",
+		address: "",
+	});
+	const [emailError, setEmailError] = useState("");
+	// This variable shows success of the request to during sign up that comes back from server
+	const [resgisterStatus, setRegisterStatus] = useState("");
+
+	const validateEmail = (email) => {
+		const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+		return emailRegex.test(email);
+	};
+
+	async function postContractorDetails(userData) {
+		try {
+			const response = await fetch("/mock-api", {
+				method: "POST",
+				headers: {
+					"Content-Type": "application/json",
+				},
+				body: JSON.stringify(userData),
+			});
+			const data = response.json();
+			if (!response.ok) {
+				setRegisterStatus(data.message);
+			}
+			setRegisterStatus(data.message);
+		} catch (error) {
+			setRegisterStatus(error.message);
+		}
+	}
+
+	const handleInputChange = (e) => {
+		const { name, value } = e.target;
+		setContractorDetails((prevDetails) => ({
+			...prevDetails,
+			[name]: value,
+		}));
+	};
+
+	const handleSubmit = (e) => {
+		e.preventDefault();
+		if (validateEmail(contractorDetails.email)) {
+			setEmailError("");
+			postContractorDetails(contractorDetails);
+		} else {
+			if (!validateEmail(contractorDetails.email)) {
+				setEmailError("Enter a valid email address");
+			}
+		}
+	};
+	function displayMessage(msg) {
+		if (msg === "successful") {
+			return "registered successfully!";
+		} else if (msg === "mismatchPassword") {
+			return "passwords do not match";
+		} else if (msg === "emailAlreadyExists") {
+			return "email exist";
+		} else if (msg === "invalidType") {
+			return "Password is incorrect!";
+		} else if (msg === "serverError") {
+			return "Internal server error";
+		} else {
+			return "Unkown error";
+		}
```

Comment:
There should only be 3 possible scenarios:
- Success
- Invalid credentials (without giving details of what part was invalid)
- Server error